### PR TITLE
Replace GC_KIND variable with SKIP_RESOURCE_CLEANUP

### PR DESCRIPTION
**Pull request by geetikabatra**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by srm09** [comment]
As part of this PR, could you also include the other flag mentioned here https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/1439

**Review by srm09 — COMMENTED** [review]
one minor nit

**Comment by srm09** [comment]
Could you include the description of the new flag in the https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/tree/master/test/e2e#running-the-e2e-tests section?

File: test/e2e/README.md
Comment (srm09) [inline test/e2e/README.md:42]: This flags skips cleanup of the resources created during the tests as well as the kind/bootstrap cluster. Could you update the docs to say the same.

File: test/e2e/README.md
Comment (srm09) [inline test/e2e/README.md:49]:
```suggestion
make e2e
```
Since the flag is set to `false` by default.

File: sigs.k8s.io/cluster-api-provider-vsphere/apis/v1alpha3/zz_generated.conversion.go
Comment (srm09) [inline sigs.k8s.io/cluster-api-provider-vsphere/apis/v1alpha3/zz_generated.conversion.go:1]: remove this file, should not be committed.

File: sigs.k8s.io/cluster-api-provider-vsphere/apis/v1alpha4/zz_generated.conversion.go
Comment (srm09) [inline sigs.k8s.io/cluster-api-provider-vsphere/apis/v1alpha4/zz_generated.conversion.go]: remove this file, should not be committed.

**Review by srm09 — CHANGES_REQUESTED** [review]
Please address the comments and squash the commits, and this is ready to merge after.

**Comment by srm09** [comment]
Do squash the commits before this can be merged.

**Comment by geetikabatra** [comment]
Done!

**Pull request merged**
